modules/dependencies.go

```diff
@@ -0,0 +1,338 @@
+package modules
+
+import (
+	"errors"
+	"io"
+	"io/ioutil"
+	"net"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/NebulousLabs/Sia/build"
+	"github.com/NebulousLabs/Sia/persist"
+	"github.com/NebulousLabs/fastrand"
+)
+
+// Dependencies defines dependencies used by all of Sia's modules. Custom
+// dependencies can be created to inject certain behavior during testing.
+type (
+	Dependencies interface {
+		// AtLeastOne will return a value that is at least one. In production,
+		// the value should always be one. This function is used to test the
+		// idempotency of actions, so during testing sometimes the value
+		// returned will be higher, causing an idempotent action to be
+		// committed multiple times. If the action is truly idempotent,
+		// committing it multiple times should not cause any problems or
+		// changes.
+		AtLeastOne() uint64
```

Comment:
it looks like this function is only used by the host to simulate performing an operation more than once. I suggest that we refactor the code to instead use `Disrupt`. We should try to keep the number of methods as small as possible.